src/thing_shots.c

```diff
@@ -661,16 +661,15 @@ long shot_hit_object_at(struct Thing *shotng, struct Thing *target, struct Coord
     HitPoints damage = 0;
     if (shotng->shot.damage)
     {
-        if (object_can_be_damaged(target)) // do not damage objects that cannot be destroyed
+        HitPoints damage_done;
+        damage_done = apply_damage_to_thing(target, shotng->shot.damage, shotst->damage_type, -1);
+        target->heart.some_countdown = 20;
+
+        // Drain allows caster to regain half of damage
+        if ((shotst->model_flags & ShMF_LifeDrain) && thing_is_creature(shootertng)) 
         {
-            damage = apply_damage_to_thing(target, shotng->shot.damage, shotst->damage_type, -1);
-            // Drain allows caster to regain half of damage, even against objects
-            if ((shotst->model_flags & ShMF_LifeDrain) && thing_is_creature(shootertng))
-            {
-                apply_health_to_thing(shootertng, shotng->shot.damage / 2);
-            }
+            apply_health_to_thing(shootertng, damage_done/2);
```

Comment:
for objects it's apply_health_to_thing and for creatures give_shooter_drained_health, you happen to know if there's a reason for that distinction?

Reply:
Seems like the same function really, but `apply_health_to_thing` seemed intended for general use and `give_shooter_drained_health` for shots. The latter has the ability though for negative health. I use `give_shooter_drained_health` in both places now.